src/lib/classes/universal-profile.spec.ts

```diff
@@ -214,40 +209,46 @@ describe('UniversalProfile', () => {
       keyManager = new ethers.Contract(LSP6KeyManager.address, KeyManagerContract.abi, provider);
     });
 
-    it('1st address should have DEFAULT_PERMISSIONS set', async () => {
-      const [signerPermissions] = await universalProfile
+    it('1st address should have ALL_PERMISSIONS set', async () => {
+      const signerPermissions = await universalProfile
         .connect(signers[0])
-        .callStatic.getDataBatch([PREFIX_PERMISSIONS + firstControllerAddress.substring(2)]);
+        .getData(
+          ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+            firstControllerAddress.substring(2)
+        );
 
-      expect(signerPermissions).toEqual(ERC725.encodePermissions(DEFAULT_PERMISSIONS));
+      expect(signerPermissions).toEqual(ALL_PERMISSIONS);
     });
 
     it('1st address should be registered in AddressPermissions[0] array', async () => {
       const hexIndex = ethers.utils.hexlify([0]);
       const key =
-        ADDRESS_PERMISSIONS_ARRAY_KEY.slice(0, 34) +
+        ERC725YDataKeys.LSP6['AddressPermissions[]'].index +
         ethers.utils.hexZeroPad(hexIndex, 16).substring(2);
 
-      const [result] = await universalProfile.connect(signers[0]).callStatic.getDataBatch([key]);
+      const result = await universalProfile.connect(signers[0]).callStatic.getData(key);
       const checkedsumResult = ethers.utils.getAddress(result);
       expect(checkedsumResult).toEqual(firstControllerAddress);
     });
 
-    it('2nd address should have DEFAULT_PERMISSIONS set', async () => {
-      const [signerPermissions] = await universalProfile
+    it('2nd address should have ALL_PERMISSIONS set', async () => {
+      const signerPermissions = await universalProfile
         .connect(signers[1])
-        .callStatic.getDataBatch([PREFIX_PERMISSIONS + secondControllerAddress.substring(2)]);
+        .getData(
+          ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+            secondControllerAddress.substring(2)
+        );
 
       expect(signerPermissions).toEqual(customPermissions);
     });
 
     it('2nd address should be registered in AddressPermissions[1] array', async () => {
       const hexIndex = ethers.utils.hexlify([1]);
       const key =
-        ADDRESS_PERMISSIONS_ARRAY_KEY.slice(0, 34) +
+        ERC725YDataKeys.LSP6['AddressPermissions[]'].index +
         ethers.utils.hexZeroPad(hexIndex, 16).substring(2);
 
-      const [result] = await universalProfile.connect(signers[0]).callStatic.getDataBatch([key]);
+      const result = await universalProfile.connect(signers[0]).getData(key);
```

Comment:
Just for my information, here we could also connect through signer[1] to the universal profile rigth ?

Reply:
Good point. Yes you could connect from any signer. I have removed it as it is calling a `view` function (doing a staticcall), so the address calling the contract is irrelevant.

Thanks for pointing this out, makes the code cleaner :)